mypy_django_plugin/transformers/querysets.py

```diff
@@ -358,19 +356,58 @@ def gather_flat_args(ctx: MethodContext) -> list[tuple[Expression | None, Proper
     arguments when their type is a TupleType with statically known items.
     """
     lookups: list[tuple[Expression | None, ProperType]] = []
+    arg_start_idx = 0
     for expr, typ, kind in zip(ctx.args[0], ctx.arg_types[0], ctx.arg_kinds[0], strict=False):
         ptyp = get_proper_type(typ)
         if kind == ARG_STAR:
             # Expand starred tuple items if statically known
             if isinstance(ptyp, TupleType):
-                for item_typ in ptyp.items:
-                    lookups.append((None, get_proper_type(item_typ)))
+                lookups.append((None, get_proper_type(ptyp.items[arg_start_idx])))
             # If not a TupleType (e.g. list/Iterable), we cannot expand statically
+            arg_start_idx += 1
             continue
         lookups.append((expr, ptyp))
     return lookups
 
 
+def check_conflicting_attr_value(
+    ctx: MethodContext, model: DjangoModel, attr_name: str, new_attrs: dict[str, MypyType] | None = None
+) -> bool:
+    """
+    Check if adding `attr_name` would conflict with existing symbols on `model`.
+
+    Args:
+        - model: The Django model being analyzed
+        - attr_name: The name of the attribute to be added
+        - fields: A mapping of field names to types currently being added to the model
+    """
+    is_conflicting_attr_value = bool(
+        # 1. Conflict with another symbol on the model.
+        # Ex:
+        #     User.objects.prefetch_related(Prefetch(..., to_attr="id"))
+        model.typ.type.get(attr_name)
+        # 2. Conflict with a previous annotation.
+        # Ex:
+        #     User.objects.annotate(foo=...).prefetch_related(Prefetch(...,to_attr="foo"))
+        #     User.objects.prefetch_related(Prefetch(...,to_attr="foo")).prefetch_related(Prefetch(...,to_attr="foo"))
+        or (model.typ.extra_attrs and attr_name in model.typ.extra_attrs.attrs)
+        # 3. Conflict with another symbol added in the current processing.
+        # Ex:
+        #     User.objects.prefetch_related(
+        #        Prefetch("groups", Group.objects.filter(name="test"), to_attr="new_attr"),
+        #        Prefetch("groups", Group.objects.all(), to_attr="new_attr"), # E: Not OK!
+        #     )
+        or (new_attrs is not None and attr_name in new_attrs)
+    )
+    if is_conflicting_attr_value:
+        ctx.api.fail(
+            f'Attribute "{attr_name}" already defined on "{model.typ.type.name}"',
```

Comment:
```suggestion
            f'Attribute "{attr_name}" already defined on "{model.typ.type.fullname}"',
```

maybe? https://github.com/python/mypy/blob/39427835ea4ab9e5735e7e5956063026d1bcc4bb/mypy/nodes.py#L3537-L3539

Reply:
I believe we would end up with the fully qualified name ie:
 `django.contrib.auth.models.User@AnnotatedWith[TypedDict({'foo': Any})].` wichi quite verbose but maybe better ? (especially for annotated models)

Reply:
It is always better :)